DDDEastAnglia/Controllers/SessionController.cs

```diff
@@ -110,7 +113,13 @@ public ActionResult Create([Bind(Exclude = "Votes")] Session session)
             if (ModelState.IsValid)
             {
                 var addedSession = sessionRepository.AddSession(session);
-                return RedirectToAction("Details", new {id = addedSession.SessionId});
+
+                UserProfile speakerProfile = userProfileRepository.GetUserProfileByUserName(User.Identity.Name);
+                
+                var sessionCreatedMailTemplate = SessionCreatedMailTemplate.Create(addedSession);
+                emailMessengerFactory.CreateEmailMessenger(sessionCreatedMailTemplate).Notify(speakerProfile);
```

Comment:
This is basically the one thing that I don't like about the design that we have for sending users messages: the email messenger exists on a per-email basis, and so needs to be called into existence before we can tell it to notify the given user. 

This could be improved by passing the `IMailTemplate` to `Notify()` as well as the speaker profile, so it becomes more like "Notify this user using this pre-determined message template", which makes sense in some respects, but this approach ties the `IMessenger` interface to email messages. A way to fix that would be to make `IMessenger` generic on the thing passed to `Notify()`, but that feels like lazy design to me. 

I think maybe I'll try hooking in Twitter notifications (or similar - texts, perhaps) and see what that does to the design of these interfaces.